### PR TITLE
Add shapefn for selu and adaptive_avgpool3d

### PR DESCRIPTION
**Pull request by JackCaoG**
Add shapefn for selu

### CODE REVIEW

File: torch/csrc/lazy/core/shape_inference.cpp [hunk 1]
```diff
@@ -977,6 +977,89 @@ std::vector<Shape> compute_shape__adaptive_avg_pool2d_backward(
...
+        "adaptive_avg_pool3d(): Expected self to have non-zero size for non-batch dimensions, "
+        "but Tensor has sizes ",
+        self.sizes(),
+        " with dimension ",
+        i,
+        " being "
+        "empty");
+  }
+  TORCH_CHECK(
+      (ndim == 4 || ndim == 5),
```
Comment (Krovatkin) [inline torch/csrc/lazy/core/shape_inference.cpp:1010]: nit: we could consider moving this up as this is one of the most inexpensive checks

File: torch/csrc/lazy/core/shape_inference.cpp [hunk 2]
```diff
@@ -977,6 +977,89 @@ std::vector<Shape> compute_shape__adaptive_avg_pool2d_backward(
...
+      "adaptive_avg_pool3d: elements of output_size must be greater than or equal to 0 ",
+      "but received {",
+      output_size[0],
+      ", ",
+      output_size[1],
+      ", ",
+      output_size[2],
+      "}");
+  int64_t ndim = self.ndimension();
+  for (const auto i : c10::irange(1, ndim)) {
```
Comment (Krovatkin) [inline torch/csrc/lazy/core/shape_inference.cpp:998]: I understand this check came from pytorch core but it looks funny. If we have a 4D tensor (CxDxHxW) wouldn't this skip over testing the channel dimension even though the comment says we should be checking all non-batch dimensions?

File: torch/csrc/lazy/core/shape_inference.cpp [hunk 3]
```diff
@@ -977,6 +977,89 @@ std::vector<Shape> compute_shape__adaptive_avg_pool2d_backward(
...
+std::vector<Shape> compute_shape__adaptive_avg_pool3d_backward(
+    const at::Tensor& grad_output,
+    const at::Tensor& self) {
+  // Checks based on `aten/src/ATen/native/AdaptiveAveragePooling.cpp`
+  int64_t ndim = grad_output.ndimension();
+
+  for (const auto i : c10::irange(1, ndim)) {
+    TORCH_CHECK(
+        grad_output.size(i) > 0,
+        "adaptive_avg_pool3d_backward(): Expected grad_output to have non-zero size for non-batch dimensions, "
```
Comment (Krovatkin) [inline torch/csrc/lazy/core/shape_inference.cpp:1040]: ditto

File: torch/csrc/lazy/core/shape_inference.cpp [hunk 4]
```diff
@@ -977,6 +977,89 @@ std::vector<Shape> compute_shape__adaptive_avg_pool2d_backward(
...
+      "adaptive_avg_pool3d_backward(): Expected 4D or 5D tensor, but got ",
+      self.sizes());
+  TORCH_CHECK(
+      self.dtype() == grad_output.dtype(),
+      "expected dtype ",
+      self.dtype(),
+      " for `grad_output` but got dtype ",
+      grad_output.dtype());
+
+  return {Shape(self.scalar_type(), self.sizes().vec())};
```
Comment (Krovatkin) [inline torch/csrc/lazy/core/shape_inference.cpp:1060]: this is probably not 100% foolproof? For example, we could pass a gradient whose spatial dimensions are bigger than those of an input?

**Review by Krovatkin — APPROVED** [review]
:shipit:

File: torch/csrc/lazy/core/shape_inference.cpp (hunk 2 above)
Comment (JackCaoG) [reply torch/csrc/lazy/core/shape_inference.cpp:998]: I guess the right thing to do here is 
```
c10::irange(ndim-4, ndim)
```

**Pull request closed without merging**